combat-trainer.lic

```diff
@@ -1287,6 +1358,30 @@ class SpellProcess
     end
   end
 
+  def check_trader_magic(game_state)
+    return unless DRStats.trader?
+    if Flags['ct-starlight-depleted'] then
+      echo "----OUT OF STARLIGHT - DELETING STARLIGHT MAGIC----"
+      @buff_spells.delete_if { |spell| @buff_spells[spell]['starlight'] }
+      @offensive_spells.delete_if do |spell|
+        if spell['starlight'] then true end
+      end
+      Flags['ct-cancel-regalia'] = true
+    end
+    if game_state.casting_regalia
+      if Flags['ct-starlight-depleted']  # In some conditions, regalia can run out of starlight partway through creation.
+        @regalia_array[game_state.swap_regalia_type].each do |item| 
+          bput("remove my #{item}", "into motes of silvery", "Remove what?", "You .*#{item}", 'The leather gauntlets slide')
+        end
+        @equipment_manager.wear_equipment_set?('standard')
```

Comment:
❓ Should we retreat first? I thought wearing armor had roundtime

Reply:
Existing swap logic doesn't have you retreat, so I didn't add it, though you're not the first person to ask about it...  From testing I've found that retreating in a full set of plate usually means I just sit there cycling retreat until the regalia wears off five-ten minutes later, but then that's only with the ~80 ranks I gained testing with regalia.  Easy enough to add an option for both normal and regalia swapping.  Do you feel like it should be opt-in or opt-out?

Reply:
I have it retreat twice immediately before any time it does the swap and it's behaving fine now.